### DISCUSSION TITLE
Benefits of using coil for bundled images

### DISCUSSION THREAD

**Discussion by ubuntudroid** [coil-kt/coil#1387] [Q&A] [answered]
If I want to show a bundled image (i.e. `R.drawable.my_image`), is there any benefit of using coil via `AsnycImage` for that instead of directly using `Image`? Especially performance-wise.

We have a rather long onboarding in our app in a ViewPager with lots of images in there and I want to remove some lags when swiping to the next page likely caused by image loading and rendering. Those images are all bundled with the app. Would using coil improve the situation in such a case?

**Comment by colinrtwhite** [comment, marked as answer]
It should; Compose's built-in`Image` decodes the image on the main thread. It's the same for `ImageView.setImageResource` when using views. `AsyncImage` moves all the work to a background thread, which will reduce UI lag.

**Reply by ubuntudroid** [reply]
Great, many thanks! 🙏
